front/src/modules/simulationResult/components/SpaceTimeChart/useGetProjectedTrainOperationalPoints.ts

```diff
@@ -88,8 +88,11 @@ const useGetProjectedTrainOperationalPoints = (
         } else {
           // If the manchette hasn't been saved, we want to display by default only
           // the waypoints with CH BV/00/'' and the path steps (origin, destination, vias)
-          operationalPointsWithUniqueIds = operationalPointsWithUniqueIds.filter((op) =>
-            op.extensions?.sncf ? isStation(op.extensions.sncf.ch) || op.weight === 100 : true
+          operationalPointsWithUniqueIds = operationalPointsWithUniqueIds.filter(
+            (op, i) =>
+              i === 0 ||
+              i === operationalPointsWithUniqueIds.length - 1 ||
+              (op.extensions?.sncf ? isStation(op.extensions.sncf.ch) || op.weight === 100 : true)
           );
```

Comment:
Hum it's a bit hard to read. Can you write a more explicit function like this ?
```suggestion
           operationalPointsWithUniqueIds = operationalPointsWithUniqueIds.filter((op, i) => {
            if (i === 0 || i === operationalPointsWithUniqueIds.length - 1) return true;
            // handle waypoints added from the map
            if (!op.extensions?.sncf) return true;
            // handle waypoints added from the pathfinding or operational points on path
            return isStation(op.extensions.sncf.ch) || op.weight === 100;
          });
```

Reply:
Done :)